include/ensmallen_bits/moead/moead.hpp

```diff
@@ -0,0 +1,284 @@
+/**
+ * @file moead.hpp
+ * @author Utkarsh Rai
+ *
+ * MOEA/D, Multi Objective Evolutionary Algorithm based on Decompositon is a
+ * multi objective optimization algorithm. It employs evolutionary algorithms,
+ * to find better solutions by iterating on the previous solutions and
+ * decomposition approaches, to convert the multi objective problem to a single
+ * objective one, to find the best Pareto Front for the given problem.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef ENSMALLEN_MOEAD_MOEAD_HPP
+#define ENSMALLEN_MOEAD_MOEAD_HPP
+
+namespace ens {
+
+/**
+ * This class implements the MOEA/D algorithm with Differential Evolution
+ * crossover. Step numbers used in different parts of the implementation
+ * correspond to the step number used in the original algorithm by the author.
+ *
+ * For more information, see the following:
+ * @code
+ * @article{article,
```

Comment:
The citations should be kept simple according to the [style guide](https://github.com/mlpack/mlpack/wiki/DesignGuidelines#citations), we don't need a bunch of fields, the author, year, pages, title, journal/conference should be sufficient!